dsgrid/query/derived_dataset.py

```diff
@@ -317,12 +326,21 @@ def _get_dimension_reference(dim_model: DimensionModel, project_config):
     return dim_ref.serialize()
 
 
-def _get_supplemental_dimension_mapping_reference(dim_model: DimensionModel, project_config):
-    key, _ = project_config.get_base_to_supplemental_config(dim_model.dimension_query_name)
+def _get_supplemental_dimension_mapping_reference(
+    dim_model: DimensionModel, project_config: ProjectConfig
+):
+    mapping_configs = project_config.list_base_to_supplemental_mapping_configs(
+        supplemental_dimension_id=dim_model.dimension_id
+    )
+    if len(mapping_configs) > 1:
+        msg = "More than one base-to-supplemental mapping config for {dim_model.label}"
+        raise NotImplementedError(msg)
```

Comment:
I haven't figured out how to handle this.

Reply:
This is called from `create_derived_dataset_config_from_query`. Can the query file or metadata file preserve information about the base dimensions included in the query?